### PR TITLE
Don't resolve image by contacting registry for docker stack deploy tests

### PR DESCRIPTION
**Pull request by nishanttotla**
This PR turns off digest pinning for `docker stack deploy` integration-cli tests.

It should be merged as soon as the CLI changes of https://github.com/docker/cli/pull/121 are merged and included.

We did a similar thing for `docker service create` and `update` in https://github.com/moby/moby/pull/33274

NOTE: The flag that's being added depends on https://github.com/docker/cli/pull/121, so I've updated the CLI commit.

### CODE REVIEW

**Comment by nishanttotla** [comment]
Also updated tests that were failing because they didn't account for the default `latest` digest added to service images.

**Comment by nishanttotla** [comment]
Ping @aaronlehmann @vdemeester @tiborvass this PR is ready now. I'm not sure if it's okay to update the CLI commit hash, but the tests should be updated (note that these fixes were already part of the cherry picks into RC2 and RC3 for 17.06).

**Comment by aaronlehmann** [comment]
LGTM

**Review by thaJeztah — CHANGES_REQUESTED** [review]
I'd like to have consensus on our approach  for bumping the CLI first; I'll see if we can discuss this ASAP

**Comment by nishanttotla** [comment]
@thaJeztah sounds good. I updated CLI here to make sure tests pass.

File: hack/dockerfile/binaries-commits [hunk 1]
```diff
@@ -12,4 +12,4 @@ BINDATA_COMMIT=a0ff2567cfb70903282db057e799fd826784d41d
 
 # CLI
 DOCKERCLI_REPO=https://github.com/docker/cli
-DOCKERCLI_COMMIT=3dfb8343b139d6342acfd9975d7f1068b5b1c3d3
+DOCKERCLI_COMMIT=4d980880f30eddec01dbe4328e0cd656a52b502b
```
Comment (tiborvass) [inline hack/dockerfile/binaries-commits]: i'm not sure about this :S

Comment (nishanttotla) [inline hack/dockerfile/binaries-commits]: @tiborvass that makes two of us.

**Review by tiborvass — CHANGES_REQUESTED** [review]
I suggest changing the scripts to install docker cli by downloading the released 17.06 static binary. Probably in a separate PR to not conflate the two problems.

**Comment by aaronlehmann** [comment]
Let's not block this PR because we want to make changes to how the CLI is installed. Without these changes to the tests, the tests won't run correctly with the released 17.06 binary anyway.

**Comment by aaronlehmann** [comment]
Has a decision been made about this one?

**Comment by dnephin** [comment]
I agree we should open a PR to pin the client at the 17.06 release version.

These tests should be deleted, not updated. They're only testing the cli, so they are not providing any value in this repo. They won't catch any regressions to the cli because they are not run as part of cli changes.

**Pull request closed without merging**